src/main/java/org/openpnp/vision/pipeline/CvPipeline.java

```diff
@@ -194,27 +233,47 @@ public Nozzle getNozzle() {
     }
   
     public void setFeeder(Feeder feeder) {
+		fiducial = false;
         this.feeder = feeder;
     }
 
     public Feeder getFeeder() {
         return feeder;
     }
+  
+    public void setPart(Part part) {
+        this.part = part;
+    }
 
-    public long getTotalProcessingTimeNs() {
-      return totalProcessingTimeNs;
+    public Part getPart() {
+        return part;
     }
 
-    public void setTotalProcessingTimeNs(long totalProcessingTimeNs) {
-      this.totalProcessingTimeNs = totalProcessingTimeNs;
+    public long getTotalProcessingTimeNs() {
+      return totalProcessingTimeNs;
     }
 
+	
+	private boolean isNull(Object obj) {
+							if(obj!=null) {
+								if(obj instanceof List<?>) {
+									if(((List)obj).size()==0) {
+										return true;
+									}
+									return false;
+								}
+								return false;	
+							}
+							return true;	
+	}
+	
     public void process() {
-
-        totalProcessingTimeNs = 0;
+        long totalProcessingTimeNs = System.nanoTime();
         release();
-        for (CvStage stage : stages) {
-            // Process and time the stage and get the result.
+		int mark=-1,last=stages.size()-1,pos=0;
+		for(int limit=1000;pos>=0&&pos<=last;pos++,limit--) {
```

Comment:
What is all of this supposed to do? Why does it need to be changed? Why is there a hardcoded value of 1000?